codec-mqtt/src/main/java/io/netty/handler/codec/mqtt/MqttEncoder.java

```diff
@@ -196,21 +223,34 @@ private static int getConnVariableHeaderFlag(MqttConnectVariableHeader variableH
     }
 
     private static ByteBuf encodeConnAckMessage(
+            ChannelHandlerContext ctx,
             ByteBufAllocator byteBufAllocator,
             MqttConnAckMessage message) {
-        ByteBuf buf = byteBufAllocator.buffer(4);
+        final MqttVersion mqttVersion = MqttCodecUtil.getMqttVersion(ctx);
+        ByteBuf propertiesBuf = encodePropertiesIfNeeded(mqttVersion,
```

Comment:
you will need to call `propertiesBuf.release()` later as you not transfer ownership